code/modules/mob/living/simple_animal/hostile/gym_rats.dm

```diff
@@ -143,49 +182,53 @@
 /mob/living/simple_animal/hostile/retaliate/gym_rat/proc/Calm()
 	enemies.Cut()
 	LoseTarget()
-	src.say(pick("Yeah, you better run.","Not worth my time, anyways."))
-	src.visible_message("<span class='notice'>[src] squeaks softly and calms down.</span>")
 
 /mob/living/simple_animal/hostile/retaliate/gym_rat/Retaliate()
 	if(!stat)
 		..()
-		src.say(pick("You want to go? Let's go!","You can't beat me, nerd!","I'll break you in half!"))
 
-/mob/living/simple_animal/hostile/retaliate/gym_rat/attackby(var/obj/item/O as obj, var/mob/user as mob) // Feed the gym rat some food
+/mob/living/simple_animal/hostile/retaliate/gym_rat/attackby(var/obj/item/weapon/reagent_containers/food/snacks/F as obj, var/mob/user as mob) // Feed the gym rat some food
 	if(stat == CONSCIOUS)
-		if(istype(O, /obj/item/weapon/reagent_containers/food/snacks/cheesewedge)) // Cheesewedges heal it a bit
-			Calm()
-			health+=5
+		if(istype(F, /obj/item/weapon/reagent_containers/food/snacks/cheesewedge_scraps)) // Cheesewedge scraps heal it a bit
+			health+=4
 			playsound(src, 'sound/items/eatfood.ogg', rand(10,50), 1)
-			visible_message("<span class='notice'>[user] feeds \the [O] to [src]. It squeaks loudly.</span>")
+			visible_message("<span class='notice'>[user] feeds \the [F] to [src]. It squeaks loudly.</span>")
 			var/image/heart = image('icons/mob/animal.dmi',src,"heart-ani2")
 			heart.plane = ABOVE_HUMAN_PLANE
 			flick_overlay(heart, list(user.client), 20)
 			flick(icon_eat, src)
-			qdel(O)
-		else if(istype(O, /obj/item/weapon/reagent_containers/food/snacks/sliceable/cheesewheel))
-			Calm()
-			health+=25
+			qdel(F)
+		if(istype(F, /obj/item/weapon/reagent_containers/food/snacks/cheesewedge)) // Cheesewedges heal it a bit
+			health+=6
 			playsound(src, 'sound/items/eatfood.ogg', rand(10,50), 1)
-			visible_message("<span class='notice'>[user] feeds \the [O] to [src]. It squeaks loudly.</span>")
+			visible_message("<span class='notice'>[user] feeds \the [F] to [src]. It squeaks loudly.</span>")
 			var/image/heart = image('icons/mob/animal.dmi',src,"heart-ani2")
 			heart.plane = ABOVE_HUMAN_PLANE
 			flick_overlay(heart, list(user.client), 20)
 			flick(icon_eat, src)
-			qdel(O)
-		else if(istype(O, /obj/item/weapon/reagent_containers/food/snacks/meat)) // Meat heals less, but packs on some extra maximum hp
-			Calm()
+			qdel(F)
+		else if(istype(F, /obj/item/weapon/reagent_containers/food/snacks/sliceable/cheesewheel))
+			health+=30
```

Comment:
you can even inherit it from here, just make a var for how much health it gains from eating and such, reduce your added code by 4x

Reply:
Let me know if the above addresses this sufficiently.